### PR TITLE
feature: namespace options and supplemental groups for cri manager

### PR DESCRIPTION
**Pull request by YaoZengzeng**
Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

### Ⅱ. Does this pull request fix one issue?

fixes part of #635 

### Ⅴ. Special notes for reviews

We skip the test of "HostNetworkMode", because CRI manager doesn't support logging now.

### CODE REVIEW

**Comment by YaoZengzeng** [comment]
@allencloud PTAL.

**Comment by Letty5411** [comment]
@YaoZengzeng could you also add a test for `group-add` flag?

**Comment by YaoZengzeng** [comment]
@allencloud @Letty5411 PTAL.

File: daemon/mgr/spec_process.go [hunk 1]
```diff
@@ -62,21 +63,30 @@ func setupProcessTTY(ctx context.Context, c *ContainerMeta, spec *SpecWrapper) e
 }
 
 func setupProcessUser(ctx context.Context, c *ContainerMeta, spec *SpecWrapper) (err error) {
+	user := specs.User{}
```
Comment (allencloud) [inline daemon/mgr/spec_process.go]: I would invite @Ace-Tang to take a look at the implementation in `setupProcessUser`.

File: daemon/mgr/spec_process.go [hunk 2]
```diff
@@ -62,21 +63,30 @@ func setupProcessTTY(ctx context.Context, c *ContainerMeta, spec *SpecWrapper) e
...
-		user.UID = uint32(uid)
-	} else {
-		user.Username = u
+	for _, group := range c.HostConfig.GroupAdd {
+		gid, err := strconv.ParseUint(group, 10, 32)
+		if err != nil {
+			logrus.Errorf("failed to parse supplemental group id %s: %v", group, err)
+			continue
+		}
+		user.AdditionalGids = append(user.AdditionalGids, uint32(gid))
```
Comment (Ace-Tang) [inline daemon/mgr/spec_process.go]: better to take `AdditionalGids` implement to package `pkg/user/`

**Comment by YaoZengzeng** [comment]
@allencloud Updated

File: pkg/user/user.go [hunk 3]
```diff
@@ -119,6 +119,22 @@ func GetIntegerID(user string) (uint32, uint32) {
 	return uint32(uid), uint32(gid)
 }
 
+// GetAdditionalGids parse supplementary gids from slice groups.
```
Comment (allencloud) [inline pkg/user/user.go]: Could you add a unit test case for this func `GetAdditionalGids`?

**Comment by allencloud** [comment]
Do you have any further thoughts on this?  @Ace-Tang

**Comment by allencloud** [comment]
LGTM

**Pull request merged**